pkgs/development/python-modules/marshmallow/2.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, buildPythonPackage, fetchPypi,
+  dateutil, simplejson, isPy27
+}:
+
+buildPythonPackage rec {
+  pname = "marshmallow";
+  version = "2.20.5";
+  disabled = isPy27;
+
+  meta = {
+    homepage = "https://github.com/marshmallow-code/marshmallow";
+    description = ''
+      A lightweight library for converting complex objects to and from
+      simple Python datatypes.
+    '';
+    license = lib.licenses.mit;
+  };
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "19yb2936ay2fc9aby4lyzscipf9gd9lk0zwjy7wm3b5j84pqj87f";
+  };
+
+  propagatedBuildInputs = [ dateutil simplejson ];
+
+  doCheck = false;
+}
```

Comment:
we cannot have different versions of the same package in python packages, it will subtly break things as either ver3 or ver2 will get loaded first, but never the right one for all cases.

If you need this just for the platformio application, please move into a marshmallow.nix alongside the application and use an override similar to aws-sam-cli

Reply:
It's better to apply the patch to platformio that I linked above.

Reply:
just read it, resolving this in favor of your suggestion